Gems/Atom/Feature/Common/Code/Source/Stars/StarsFeatureProcessor.cpp

```diff
@@ -0,0 +1,294 @@
+            m_starsVertexBuffer = RPI::BufferSystemInterface::Get()->CreateBufferFromCommonPool(desc);
+        }
+        else
+        {
+            if (m_starsVertexBuffer->GetBufferSize() != bufferSize)
+            {
+                m_starsVertexBuffer->Resize(bufferSize);
+            }
+
+            m_starsVertexBuffer->UpdateData(m_starsMeshData.data(), bufferSize);
+        }
+
+        m_meshStreamBufferViews[0] = RHI::StreamBufferView(*m_starsVertexBuffer->GetRHIBuffer(), 0, bufferSize, elementSize);
+
+        UpdateDrawPacket();
+    }
+
+    void StarsFeatureProcessor::SetExposure(float exposure)
+    {
+        m_exposure = exposure;
+        m_updateShaderConstants = true;
+    }
+
+    void StarsFeatureProcessor::SetRadiusFactor(float radiusFactor)
+    {
+        m_radiusFactor = radiusFactor;
+        m_updateShaderConstants = true;
+    }
+
+    void StarsFeatureProcessor::SetOrientation(AZ::Quaternion orientation)
+    {
+        m_orientation = AZ::Matrix3x3::CreateFromQuaternion(orientation);
+        m_updateShaderConstants = true;
+    }
+
+    void StarsFeatureProcessor::SetTwinkleRate(float twinkleRate)
+    {
+        m_shaderConstants.m_twinkleRate = twinkleRate;
+        m_updateShaderConstants = true;
+    }
+
+    void StarsFeatureProcessor::OnRenderPipelineAdded([[maybe_unused]] RPI::RenderPipelinePtr renderPipeline)
+    {
+        if(!m_meshPipelineState)
+        {
+            m_meshPipelineState = aznew RPI::PipelineStateForDraw;
+            m_meshPipelineState->Init(m_shader);
+
+
+            RHI::InputStreamLayoutBuilder layoutBuilder;
+            layoutBuilder.AddBuffer()
+                ->Channel("POSITION", RHI::Format::R32G32B32_FLOAT)
+                ->Channel("COLOR", RHI::Format::R8G8B8A8_UNORM);
+            layoutBuilder.SetTopology(RHI::PrimitiveTopology::TriangleList);
+            auto inputStreamLayout = layoutBuilder.End();
+
+            m_meshPipelineState->SetInputStreamLayout(inputStreamLayout);
+            m_meshPipelineState->SetOutputFromScene(GetParentScene());
+            m_meshPipelineState->Finalize();
+
+            UpdateDrawPacket();
+            UpdateBackgroundClearColor();
+        }
+    }
+
+    void StarsFeatureProcessor::OnRenderPipelinePassesChanged([[maybe_unused]] RPI::RenderPipeline* renderPipeline)
+    {
+        if(m_meshPipelineState)
+        {
+            m_meshPipelineState->SetOutputFromScene(GetParentScene());
+            m_meshPipelineState->Finalize();
+
+            UpdateDrawPacket();
+            UpdateBackgroundClearColor();
+        }
+    }
+
+    void StarsFeatureProcessor::OnViewportSizeChanged(AzFramework::WindowSize size)
+    {
+        m_viewportSize = size;
+        m_updateShaderConstants = true;
+    }
+
+    void StarsFeatureProcessor::OnAssetReloaded([[maybe_unused]] Data::Asset<Data::AssetData> asset)
+    {
+        UpdateDrawPacket();
+    }
+
+    void StarsFeatureProcessor::UpdateBackgroundClearColor()
```

Comment:
Suggestion: I think the design of having stars over a black background is too specific. 
I think a better idea could be render them after skybox (could be in the end of skybox render pass).

Reply:
You may try to render it to transparent pass which is rendered after skybox pass and composite pass. All you need to do is to change the value of 'DrawlistTag' from 'forward' to 'transparent' in the .shader file.

Reply:
I experimented with this but now I'm concerned about the situation where you want to see stars through a tinted transparent window (or transparent particle like smoke), in this case I want the stars to be additively drawn on top of the Skybox, but not on top of the tinted window or particle.